### PR TITLE
fix: do not allow `SequenceLocation` to have negative `start`/`end` v…

### LINKED ISSUE #442: SequenceLocation allows for negative start/end coordinates
https://github.com/ga4gh/vrs/blob/2.x/schema/vrs/json/SequenceLocation#L77

https://github.com/ga4gh/vrs/blob/2.x/schema/vrs/json/SequenceLocation#L88

We do not do any validation at the moment. We should add validation checks to follow what is stated in the descriptions.

### PR DESCRIPTION
**Pull request by korikuzma**
…alues

close #442

* Add model_validators to `Range` and `SequenceLocation`

### CODE REVIEW

File: src/ga4gh/vrs/models.py [hunk 1]
```diff
@@ -454,15 +471,55 @@ class SequenceLocation(_Ga4ghIdentifiableObject):
     )
     start: Optional[Union[Range, int]] = Field(
         None,
-        description='The start coordinate or range of the SequenceLocation. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range less than the value of `end`.',
+        description='The start coordinate or range of the SequenceLocation. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range less than or equal to the value of `end`.',
```
Comment (ahwagner) [inline src/ga4gh/vrs/models.py:474]: Like that this needs to be >= 0. There should not be a constraint on `start` < `end` for VRS 2.0, but just noted that this wasn't updated in the VRS 2.0 field definitions. Will address presently.

File: src/ga4gh/vrs/models.py [hunk 2]
```diff
@@ -454,15 +471,55 @@ class SequenceLocation(_Ga4ghIdentifiableObject):
...
     )
     end: Optional[Union[Range, int]] = Field(
         None,
-        description='The end coordinate or range of the SequenceLocation. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range greater than the value of `start`.',
+        description='The end coordinate or range of the SequenceLocation. The minimum value of this coordinate or range is 0. MUST represent a coordinate or range greater than or equal to the value of `start`.',
```
Comment (ahwagner) [inline src/ga4gh/vrs/models.py:478]: Like that this needs to be >= 0. There should not be a constraint on `start` < `end` for VRS 2.0, but just noted that this wasn't updated in the VRS 2.0 field definitions. Will address presently.

**Review by ahwagner — COMMENTED** [review]
start < end should not be enforced, unless `SequenceReference.circular = False`. Though it would be nice to raise a warning if end < start and SequenceReference.circular is undefined. A discussion on this topic was first raised in #362, and later moved ga4gh/vrs#477. The VRS documentation doesn't capture this (ope), but it will be necessary to represent variants spanning the origin.

This isn't described in the VRS documentation, opened ga4gh/vrs#537 to address. Just providing a heads-up here as this PR contains some validation code corresponding to this issue.

**Comment by korikuzma** [comment]
@ahwagner can you re-review?

**Review by ahwagner — APPROVED** [review]

**Pull request merged**